native/src/routes/Settings.tsx

```diff
@@ -31,36 +29,25 @@ const SectionHeader = styled.Text`
 `
 
 const Settings = ({ navigation }: SettingsProps): ReactElement => {
-  const { settings, updateSettings } = useAppContext()
-  const { cityCode, languageCode } = useContext(AppContext)
+  const appContext = useCityAppContext()
   const showSnackbar = useSnackbar()
   const { t } = useTranslation('settings')
+  const { settings } = appContext
 
-  const setSetting = async (
-    changeSetting: (settings: SettingsType) => Partial<SettingsType>,
-    changeAction?: (settings: SettingsType) => Promise<boolean>,
-  ) => {
+  const safeOnPress = (update: () => Promise<void> | void) => () => {
     const oldSettings = settings
-    const newSettings = { ...oldSettings, ...changeSetting(settings) }
-    updateSettings(newSettings)
-
-    try {
-      const successful = changeAction ? await changeAction(newSettings) : true
-
-      if (!successful) {
-        updateSettings(oldSettings)
-      }
-    } catch (e) {
+    update()?.catch(e => {
```

Comment:
:wrench: : I am not quite sure what you want to achive, but this will only handle errors that are from async functions. If a update function is synchronous it will not be handled (like in a try/catch block). This may be no problem currently as the synchronous function do not need this error handling, bu I think it may be hard to understand / maintain. So maybe we can either also catch also errors from synchronous functions or rename this function so at least the naming makes clear this only safes async functions.